coderd/metricscache/metricscache_test.go

```diff
@@ -47,6 +47,27 @@ func newMetricsCache(t *testing.T, log slog.Logger, clock quartz.Clock, interval
 	return cache, db
 }
 
+func awaitCacheRefresh(ctx context.Context, t *testing.T, clock *quartz.Mock) chan struct{} {
+	t.Helper()
+	ch := make(chan struct{})
+	trapNow := clock.Trap().Now()
```

Comment:
Setting of the traps races with the function you are trying to test. You need to set your traps _before_ you call the function under test.